constrainedfilefield/static/constrainedfilefield/js/file_checker.js

```diff
@@ -1,3 +1,20 @@
+function humanReadableSize(bytes) {
+    units = ['Bytes', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB']
+    var initialbytes = bytes
+
+    var i = 0
+    while (bytes >= 1024) {
+        bytes /= 1024
+        i++
+    }
+    
+    if (i > units.length) {
+        return initialbytes + ' ' + units[0] 
```

Comment:
I think it would make more sense to return the expressed value in the largest unit instead. For example, something like: `return (~~(initialbytes/(1024*(units.length-1)))).toFixed(2) + ' ' + units[units.length-1]`.
Even though this might never happen, at least I am not aware of any server supporting such large input files ;)